src/github/SyftGithubAction.ts

```diff
@@ -0,0 +1,398 @@
+        },
+        debug(message) {
+          core.debug(message);
+        },
+      },
+    });
+
+    if (exitCode > 0) {
+      error = new Error("An error occurred running Syft");
+    } else {
+      return stdout;
+    }
+  } catch (e) {
+    error = e;
+  }
+
+  core.error(stdout);
+  core.error(stderr);
+  throw error;
+}
+
+/**
+ * Downloads the appropriate Syft binary for the platform
+ */
+export async function downloadSyft(): Promise<string> {
+  const name = SYFT_BINARY_NAME;
+  const version = SYFT_VERSION;
+
+  const url = `https://raw.githubusercontent.com/anchore/${name}/main/install.sh`;
+
+  core.debug(`Installing ${name} ${version}`);
+
+  // Download the installer, and run
+  const installPath = await cache.downloadTool(url);
+
+  // Make sure the tool's executable bit is set
+  await exec.exec(`chmod +x ${installPath}`);
+
+  const cmd = `${installPath} -b ${installPath}_${name} ${version}`;
+  await exec.exec(cmd);
+
+  return `${installPath}_${name}/${name}`;
+}
+
+/**
+ * Gets the Syft command to run via exec
+ */
+export async function getSyftCommand(): Promise<string> {
+  const name = SYFT_BINARY_NAME;
+  const version = SYFT_VERSION;
+
+  let syftBinary = cache.find(name, version);
+  if (!syftBinary) {
+    // Not found; download and install it
+    syftBinary = await downloadSyft();
+
+    // Cache the downloaded file
+    syftBinary = await cache.cacheFile(syftBinary, name, name, version);
+  }
+
+  // Add tool to path for this and future actions to use
+  core.addPath(syftBinary);
+  return name;
+}
+
+/**
+ * Returns the SBOM format as specified by the user, defaults to SPDX
+ */
+export function getSbomFormat(): SyftOptions["format"] {
+  return (core.getInput("format") as SyftOptions["format"]) || "spdx";
+}
+
+/**
+ * Uploads a SBOM as a workflow artifact
+ * @param contents SBOM file contents
+ */
+export async function uploadSbomArtifact(contents: string): Promise<void> {
+  const { repo } = github.context;
+  const client = getClient(repo, core.getInput("github_token"));
+
+  const fileName = getArtifactName();
+
+  const tempPath = fs.mkdtempSync(path.join(os.tmpdir(), "sbom-action-"));
+  const filePath = `${tempPath}/${fileName}`;
+  fs.writeFileSync(filePath, contents);
+
+  const outputFile = core.getInput("output_file");
+  if (outputFile) {
+    fs.copyFileSync(filePath, outputFile);
+  }
+
+  core.info(dashWrap(`Uploading workflow artifacts`));
+  core.info(`Artifact: ${filePath}`);
+
+  await client.uploadWorkflowArtifact({
+    file: filePath,
+    name: fileName,
+  });
+}
+
+/**
+ * Gets a boolean input value if supplied, otherwise returns the default
+ * @param name name of the input
+ * @param defaultValue default value to return if not set
+ */
+function getBooleanInput(name: string, defaultValue: boolean): boolean {
+  const val = core.getInput(name);
+  if (val === "") {
+    return defaultValue;
+  }
+  return Boolean(val);
+}
+
+export async function runSyftAction(): Promise<void> {
```

Comment:
nit: This function is also on the large side. If there's a nice way to decompose that, I think that'd help increase readability and ease of maintenance.

Reply:
I've split this up a bit. I was experimenting a bit last night 😬